server/src/main/java/org/apache/cassandra/sidecar/tasks/CassandraClusterSchema.java

```diff
@@ -0,0 +1,273 @@
+    private static final Logger LOGGER = LoggerFactory.getLogger(CassandraClusterSchema.class);
+
+    private final AtomicReference<String> currSchemaText = new AtomicReference<>("");
+    private final AtomicReference<Set<CqlTable>> cdcTables = new AtomicReference<>(Collections.emptySet());
+    private final ConcurrentHashMap<TableIdentifier, UUID> tableIdCache = new ConcurrentHashMap<>();
+    private final CdcDatabaseAccessor databaseAccessor;
+    private final CopyOnWriteArrayList<Runnable> schemaChangeListeners = new CopyOnWriteArrayList<>();
+    private final SidecarConfiguration sidecarConfiguration;
+    private final InstanceMetadataFetcher instanceFetcher;
+    private final SecondBoundConfiguration tableSchemaRefreshTime;
+    private final CassandraBridgeFactory cassandraBridgeFactory;
+
+    public CassandraClusterSchema(InstanceMetadataFetcher instanceFetcher,
+                                  CdcDatabaseAccessor databaseAccessor,
+                                  SidecarConfiguration sidecarConfiguration,
+                                  CassandraBridgeFactory cassandraBridgeFactory)
+    {
+
+        this.instanceFetcher = instanceFetcher;
+        this.databaseAccessor = databaseAccessor;
+        this.sidecarConfiguration = sidecarConfiguration;
+        this.tableSchemaRefreshTime = sidecarConfiguration.serviceConfiguration().cdcConfiguration().tableSchemaRefreshTime();
+        this.cassandraBridgeFactory = cassandraBridgeFactory;
+    }
+
+    public void addSchemaChangeListener(Runnable listener)
+    {
+        schemaChangeListeners.add(listener);
+    }
+
+    public void refresh()
+    {
+        NodeSettings nodeSettings = instanceFetcher.callOnFirstAvailableInstance(instance-> instance.delegate().nodeSettings());
+        CassandraBridge cassandraBridge = cassandraBridgeFactory.get(nodeSettings.releaseVersion());
+        CdcBridge cdcBridge = CdcBridgeFactory.getCdcBridge(cassandraBridge);
+
+        try
+        {
+            LOGGER.info("Checking for schema changes...");
+            String fullSchemaText = databaseAccessor.fullSchema();
+            if (!fullSchemaText.equals(currSchemaText.get()))
+            {
+                LOGGER.info("Schema change detected, refreshing CDC tables");
+                currSchemaText.set(fullSchemaText);
+                Set<CqlTable> updatedCdcTables = buildCdcTables(fullSchemaText, databaseAccessor, tableIdCache, instanceFetcher, cassandraBridgeFactory);
+                LOGGER.info("Cdc enabled tables tables='{}'", 
+                            updatedCdcTables.stream()
+                                            .map(m -> String.format("%s.%s", m.keyspace(), m.table()))
+                                            .collect(Collectors.joining(",")));
+                cdcTables.set(updatedCdcTables);
+                cdcBridge.updateCdcSchema(updatedCdcTables, databaseAccessor.partitioner(),
```

Comment:
I don't think this `databaseAccessor.partitioner()` is required at all. We already have the partitioner information as a string in `nodeSettings.partitioner()`, with the caveat that we will need to check if the string has `.` and if it does we will need to do the logic we have inside `databaseAccessor.partitioner()`
```suggestion
                cdcBridge.updateCdcSchema(updatedCdcTables, Partitioner.valueOf(nodeSettings.partitioner()),
```

Reply:
and of course Partitioner is in spark, so we wouldn't be able to change it here. So we'll need to do the check before calling valueOF